File: app/src/main/java/com/example/taxconnect/model/TransactionModel.java
package com.example.taxconnect.model;

import com.google.firebase.Timestamp;

public class TransactionModel {
    private String transactionId;
    private String userId;
    private String caId;
    private String caName;
    private String description;
    private double amount;
    private String status; // "SUCCESS", "FAILED", "PENDING"
    private Timestamp timestamp;

    public TransactionModel() { }

    public TransactionModel(String transactionId, String userId, String caId, String caName, String description, double amount, String status) {
        this.transactionId = transactionId;
        this.userId = userId;
        this.caId = caId;
        this.caName = caName;
        this.description = description;
        this.amount = amount;
        this.status = status;
        this.timestamp = Timestamp.now();
    }

    public String getTransactionId() { return transactionId; }
    public void setTransactionId(String transactionId) { this.transactionId = transactionId; }

    public String getUserId() { return userId; }
    public void setUserId(String userId) { this.userId = userId; }

    public String getCaId() { return caId; }
    public void setCaId(String caId) { this.caId = caId; }

    public String getCaName() { return caName; }
    public void setCaName(String caName) { this.caName = caName; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public double getAmount() { return amount; }
    public void setAmount(double amount) { this.amount = amount; }

    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }

    public Timestamp getTimestamp() { return timestamp; }
    public void setTimestamp(Timestamp timestamp) { this.timestamp = timestamp; }
}

File: tax/TAX-connect-prototype/tax/app/src/main/java/com/example/taxconnect/data/models/TransactionModel.kt
package com.example.taxconnect.data.models

import com.google.firebase.Timestamp

data class TransactionModel(
    var transactionId: String? = null,
    var userId: String? = null,
    var caId: String? = null,
    var caName: String? = null,
    var description: String? = null,
    var amount: Double = 0.0,
    var status: String? = null, // "SUCCESS", "FAILED", "PENDING"
    var timestamp: Timestamp? = null,
    var razorpayPaymentId: String? = null,
    
    // Enhanced Fields for better querying
    var type: String? = null, // "WALLET_DEPOSIT", "WALLET_WITHDRAWAL", "SERVICE_PAYMENT", "SERVICE_INCOME"
    var userName: String? = null, // Denormalized for easy display
    var userEmail: String? = null // Denormalized for easy search
) {
    constructor(transactionId: String, userId: String, caId: String, caName: String, description: String, amount: Double, status: String) : this(
        transactionId = transactionId,
        userId = userId,
        caId = caId,
        caName = caName,
        description = description,
        amount = amount,
        status = status,
        timestamp = Timestamp.now(),
        razorpayPaymentId = null,
        type = "GENERAL",
        userName = null,
        userEmail = null
    )
}
